crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -318,6 +342,16 @@ fn build_new_replicated_loglet_configuration(
         })
         .collect();
 
+    let sequencer = preferred_sequencer
+        .and_then(|node_id| {
+            // map to a known alive node
+            observed_cluster_state.alive_nodes.get(&node_id.id())
```

Comment:
Don't we lose strictness of preference by potentially swapping the generational node id that was requested with whatever generation that we have in the list of alive nodes?

Is that intention here to fallback back in this order?
- Generational node if it's alive
- Other generation of the same node if alive
- Any other alive node

Reply:
The problem is what if a `GenerationalNodeId` is specified that is not known to be alive? In this case, the logs controller will try to reconfigure the log on the next tick.

Right now, the scheduler won't give a `GenerationalNodeId` hint since the chosen partition processor leader is not bound to a generation.